src/sentry/utils/snuba_rpc.py

```diff
@@ -61,18 +63,46 @@ def meta(
     ): ...
 
 
-def table_rpc(req: TraceItemTableRequest) -> TraceItemTableResponse:
-    resp = _make_rpc_request("EndpointTraceItemTable", "v1", req.meta.referrer, req)
-    response = TraceItemTableResponse()
-    response.ParseFromString(resp.data)
-    return response
+def table_rpc(requests: list[TraceItemTableRequest]) -> list[TraceItemTableResponse]:
+    return make_rpc_requests(table_requests=requests)[0]
 
 
-def timeseries_rpc(req: TimeSeriesRequest) -> TimeSeriesResponse:
-    resp = _make_rpc_request("EndpointTimeSeries", "v1", req.meta.referrer, req)
-    response = TimeSeriesResponse()
-    response.ParseFromString(resp.data)
-    return response
+def timeseries_rpc(requests: list[TimeSeriesRequest]) -> list[TimeSeriesResponse]:
+    return make_rpc_requests(timeseries_requests=requests)[1]
+
+
+def make_rpc_requests(
+    table_requests: list[TraceItemTableRequest] | None = None,
+    timeseries_requests: list[TimeSeriesRequest] | None = None,
+) -> tuple[list[TraceItemTableResponse], list[TimeSeriesResponse]]:
+    """Given lists of requests batch and run them together"""
+    # Throw the two lists together, _make_rpc_requests will just run them all
+    table_requests = [] if table_requests is None else table_requests
+    timeseries_requests = [] if timeseries_requests is None else timeseries_requests
+    requests = table_requests + timeseries_requests
+
+    endpoint_names = [
+        "EndpointTraceItemTable" if isinstance(req, TraceItemTableRequest) else "EndpointTimeSeries"
+        for req in requests
+    ]
+    resp = _make_rpc_requests(
+        endpoint_names, "v1", [req.meta.referrer for req in requests], requests
+    )
+
+    # Split the results back up, _make_rpc_requests will return them back in order so we can use the type in the
+    # requests list to determine which request goes where
+    timeseries_results = []
+    table_results = []
+    for request, item in zip(requests, resp):
+        if isinstance(request, TraceItemTableRequest):
+            table_response = TraceItemTableResponse()
+            table_response.ParseFromString(item.data)
+            table_results.append(table_response)
+        elif isinstance(request, TimeSeriesRequest):
+            timeseries_response = TimeSeriesResponse()
+            timeseries_response.ParseFromString(item.data)
+            timeseries_results.append(timeseries_response)
+    return table_results, timeseries_results
```

Comment:
I think if we add more types of requests to this method i'll split this out to a dict or nameddict instead of relying on tuple order